### PR TITLE
Rework homepage frontend

### PR DESCRIPTION
**Pull request by Hitansh-Shah**

### CODE REVIEW

File: apps/frontend/static_src/scss/components/doc-mode.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.doc-mode:hover {
```
Comment (allcaps) [inline apps/frontend/static_src/scss/components/doc-mode.scss:1]: We should align the name of this component with the streamfield block name. If we use `.doc-mode` the streamfield block must be `DocModeBlock`. If we go for `SectionGridBlock` this must be `.section-grid`

File: apps/frontend/static_src/scss/components/doc-mode.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    .doc-mode__container {
+        .doc-mode__row1 {
```
Comment (allcaps) [inline apps/frontend/static_src/scss/components/doc-mode.scss:3]: line 2-3 do not add styling, we remove them. Same for the HTML,if we don't apply styling, there shoudn't be class names.

File: apps/frontend/static_src/scss/main.scss [hunk 3]
```diff
@@ -44,7 +42,10 @@ $primary: $green;
 //@import "~bootstrap/scss/placeholders";
 
 // Helpers
-//@import "~bootstrap/scss/helpers";
+// @import "~bootstrap/scss/helpers";
```
Comment (allcaps) [inline apps/frontend/static_src/scss/main.scss:45]: Don't change lines that you don't want to change.

File: apps/frontend/static_src/scss/variables.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Wagtail Guide conf.
+$enable-rounded: false;
+$green:   teal !default;
+$primary: $green;
+
+$body-bg: #EDEAF3;
```
Comment (allcaps) [inline apps/frontend/static_src/scss/variables.scss:6]: Nitpick, follow the convention of Bootstrap with lower case values.

We might look into a linter for frontend code. (for a later PR)

File: apps/home/blocks.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    class Meta:
+        template = "home/section_block.html"
+
+
+class SectionGridBlock(blocks.ListBlock):
+    def __init__(self, *args, **kwargs):
+        kwargs.update({"child_block": SectionBlock(label=_("Section"))})
+        super().__init__(*args, **kwargs)
+
```
Comment (allcaps) [inline apps/home/blocks.py:26]: Can we add a get context here?
```
    def get_context(self, value, parent_context=None):
        context = super().get_context(value, parent_context)
        context["icon_template"] = f"/path/to/{self.bla}.svg"
        return context
        ```
        
to replace `apps/home/templates/home/section_block.html`  line 4.

File: apps/home/templates/svg/explanation.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<svg class="svg_icon" width="114" height="100" viewBox="0 0 114 100" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (allcaps) [inline apps/home/templates/svg/explanation.html:1]: Can we keep the file names matching the content? If we use `explanation.svg` desktop apps can open the file.

FYI, Django templating doesn't care about the filename.

Comment (allcaps) [reply apps/home/templates/svg/explanation.html:1]: Did we optimise the SVGs? https://jakearchibald.github.io/svgomg/

File: apps/home/templates/home/home_page.html [hunk 7]
```diff
@@ -1,11 +1,67 @@
...
+        Using Wagtail: an Editor's guide
+      </div>
+    </div>
+
+    <div class="row mt-5">
+      <div class="col-lg-8 mx-5">
+        <p class="fs-5">
+          This section of the documentation is written for the users of a Wagtail-powered
+          site. That is, the content editors, moderators and admnistrators who will be
+          running things on a day-to-day basis.
```
Comment (allcaps) [inline apps/home/templates/home/home_page.html:41]: If it is content, it should be editable via the Wagtail admin.

If it is the user interface (words like 'submit', 'search', 'close') it should be marked for translation. Eg `{% trans 'submit' %}`

**Review by allcaps — CHANGES_REQUESTED** [review]
Hey Hitansh,

This is much, much better than previous PR. Well done.

Most of the feedback are minor comments and stylistic things. Almost there!

File: apps/home/templates/home/home_page.html (hunk 7 above)
Comment (allcaps) [reply apps/home/templates/home/home_page.html:41]: Let's leave this ^ and the text `Using Wagtail: an Editor's guide` for now. We can pick it up later in a separate PR.

File: apps/home/blocks.py (hunk 5 above)
Comment (Hitansh-Shah) [reply apps/home/blocks.py:26]: I guess we cannot access class attributes like `self.section`, `self.title` etc. inside methods in `StructBlock`.  Here is a issue I found with same problem https://github.com/wagtail/wagtail/issues/4387. We can extend `StructValue` and define a method there. I don't know if we can define `get_context` method this way.

File: apps/home/templates/svg/explanation.html (hunk 6 above)
Comment (Hitansh-Shah) [reply apps/home/templates/svg/explanation.html:1]: Ohh nice. I didn't know that we could use `{% include svg/explanation.svg %}`. Fine I will rename `templates/svg/*.html` to `templates/svg/*.svg`. 
Also I thought that we weren't sure of optimisations yet and therefore I didn't optimise. I will optimise and include in next commit.

File: apps/home/templates/home/home_page.html (hunk 7 above)
Comment (Hitansh-Shah) [reply apps/home/templates/home/home_page.html:41]: Sure. I will keep the note about translation and content in mind for the rest of the commits.

File: apps/frontend/static_src/scss/main.scss [hunk 8]
```diff
@@ -16,13 +14,13 @@ $primary: $green;
...
-//@import "~bootstrap/scss/grid";
+@import "~bootstrap/scss/grid";
 //@import "~bootstrap/scss/tables";
 //@import "~bootstrap/scss/forms";
 @import "~bootstrap/scss/buttons";
 //@import "~bootstrap/scss/transitions";
-//@import "~bootstrap/scss/dropdown";
-//@import "~bootstrap/scss/button-group";
+@import "~bootstrap/scss/dropdown";
+// @import "~bootstrap/scss/button-group";
```
[the remark this answers is not in the document]
Comment (Hitansh-Shah) [reply apps/frontend/static_src/scss/main.scss:23]: Sorry about that 😅, I was trying to include `button-group` for some reason but then later I commented it out again. Git must have marked it as modified in the process.

File: apps/home/templates/home/home_page.html [hunk 9]
```diff
@@ -1,11 +1,67 @@
...
+        </p>
+      </div>
+    </div>
+
+    <div class="row mt-5 mx-5">
+      {% for section in page.sections %}
```
[the remark this answers is not in the document]
Comment (Hitansh-Shah) [reply apps/home/templates/home/home_page.html:47]: Sure I will do that.

File: apps/frontend/static_src/scss/components/doc-mode.scss (hunk 1 above)
Comment (Hitansh-Shah) [reply apps/frontend/static_src/scss/components/doc-mode.scss:1]: :+1:

File: apps/frontend/static_src/scss/variables.scss (hunk 4 above)
Comment (Hitansh-Shah) [reply apps/frontend/static_src/scss/variables.scss:6]: I didn't understand this one. Can you point me to that section of bootstrap docs?

**Review by allcaps — APPROVED** [review]
Cool, thank you.

Especially the `value_class = SectionStructValue` 👍 

I've fixed 2 typos and the padding/margin issue on mobile.

**Pull request merged**